lib/sources.nix

```
@@ -229,6 +229,42 @@ let
 
   canCleanSource = src: src ? _isLibCleanSourceWith || !(pathHasContext (toString src));
 
+  /*
+    Get all files (recursively) that pass a filter function. Unlike other source
+    filtering functions, this will not include empty directories in the final
+    derivation.
+
+    Type: sourceLike -> (String -> Bool) -> Source
+
+    Example:
+      filterFiles ./. (lib.hasSuffix ".html")
+  */
+  filterFiles = src: filter:
+    let
+      includedPaths =
+        builtins.filter
+          filter
+          (lib.filesystem.listFilesRecursive src);
+
+      name =
+        # Fix the problem of store paths names causing a rebuild every time they change.
+        let baseName = builtins.baseNameOf src; in
+        if builtins.match "[a-z0-9]{32}.*" baseName == null
```

```suggestion
        let baseName = builtins.baseNameOf src; in
        if lib.isStorePath src
```

(when moving it elsewhere)